evaluate/evaluate.go

```diff
@@ -46,9 +66,9 @@ func Evaluate(ctx *Context) (assessments report.AssessmentPerModelPerLanguagePer
 	assessments = report.NewAssessmentPerModelPerLanguagePerRepository(ctx.Models, ctx.Languages, ctx.RepositoryPaths)
 	problemsPerModel := map[string][]error{}
 	{
-		for r := uint(0); r < ctx.Runs; r++ {
-			if ctx.Runs > 1 {
-				ctx.Log.Printf("Run %d/%d", r+1, ctx.Runs)
+		for rl := uint(0); rl < ctx.runsAtLanguageLevel(); rl++ {
```

Comment:
I guess I found a bug with the `TemporaryRepository`. We want to copy the repository for each language only 1 time but currently we are doing it on every run. So 3 runs = 3 x language copies.

We need to iterate over the languages before this loop and create the temporaryRepositoryPaths. We then need to store them inside a map and use the `repositoryPath` as index so we can fetch the correct temporary one during the model run.
The cleanup functions also need to be taken care of in a slice/map and run after everything is done.

Reply:
This is also true for the loop below where we are not using the `RepositoryPlainName`